tests/backup/backup_helper.go

```diff
@@ -3445,9 +3447,56 @@ func DeleteAppNamespace(namespace string) error {
 		return "", false, nil
 	}
 	_, err = task.DoRetryWithTimeout(namespaceDeleteCheck, namespaceDeleteTimeout, jobDeleteRetryTime)
+	if err != nil {
+		ns, err = k8sCore.GetNamespace(namespace)
+		if err != nil {
+			return err
+		}
+		if ns.Status.Phase == "Terminating" {
+			log.Infof("Namespace - %s is in %s phase ", namespace, ns.Status.Phase)
+			err = DeleteTerminatingNamespace(namespace)
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteTerminatingNamespace deletes the given namespace which has been in terminating state
+func DeleteTerminatingNamespace(namespace string) error {
+	k8sCore := core.Instance()
+	ns, err := k8sCore.GetNamespace(namespace)
 	if err != nil {
 		return err
 	}
+	if ns.Status.Phase == "Terminating" {
+		log.Infof("Namespace - %s is in %s phase ", namespace, ns.Status.Phase)
+		ns.Spec.Finalizers = nil
+		ns, err = k8sCore.UpdateNamespace(ns)
+		if err != nil {
+			return err
+		}
+		namespaceDeleteCheck := func() (interface{}, bool, error) {
+			nsObj, err := core.Instance().GetNamespace(namespace)
+			if err != nil {
+				if strings.Contains(err.Error(), "not found") {
+					log.Infof("Namespace - %s is not found and hence deleted", namespace)
+					return "", false, nil
+				} else {
+					return "", false, err
```

Comment:
Is it `return "", true, err`?

Reply:
I do not want to retry anymore here. If there is some error in getting namespace object and the error is not "namespace not found", then there is something that has gone wrong fundamentally. It is better to bail out with the error than retrying.